scss/_patterns_side-navigation.scss

```diff
@@ -143,42 +147,69 @@
   // space for the right hand icon with default inner spacing on the left
   $sp-sidenav--icon-spacing-width: map-get($icon-sizes, default) + $sph-inner;
 
-  .p-side-navigation > .p-side-navigation__list:first-of-type,
-  .p-side-navigation__drawer > .p-side-navigation__list:first-of-type {
-    // nudge from top to put text on baseline grid
-    padding-top: 0.5 * $spv-inner--x-small;
-  }
-
-  .p-side-navigation__list {
+  %side-navigation__list {
     @extend %vf-list;
+    @extend %vf-pseudo-border--bottom;
 
-    & + & {
-      @extend %vf-pseudo-border--top;
+    &::after {
+      @include vf-side-navigation-spacing-left($prop: left);
 
-      &::after {
-        @include vf-side-navigation-spacing-left($prop: left);
+      bottom: -0.5 * $spv-outer--scaleable; // place border in the middle of the margin
+    }
 
-        top: -0.5 * $spv-outer--scaleable; // place border in the middle of the margin
-      }
+    .p-side-navigation--icons &::after {
+      @include vf-side-navigation-spacing-left($prop: left, $offset: $sp-sidenav--icon-spacing-width);
+    }
 
-      .p-side-navigation--icons &::after {
-        @include vf-side-navigation-spacing-left($prop: left, $offset: $sp-sidenav--icon-spacing-width);
-      }
+    &:last-of-type::after {
+      content: none;
     }
   }
 
+  .p-side-navigation__list {
+    @extend %side-navigation__list;
+  }
+
+  .p-side-navigation > .p-side-navigation__list:first-of-type,
+  .p-side-navigation__drawer > .p-side-navigation__list:first-of-type {
+    // nudge from top to put text on baseline grid
+    padding-top: 0.5 * $spv-inner--x-small;
+  }
+
   .p-side-navigation__item--title {
     font-weight: $font-weight-bold;
   }
 
-  .p-side-navigation__text,
-  .p-side-navigation__link {
+  %side-navigation__text {
     @include vf-side-navigation-spacing-left;
 
     display: flex;
     padding-bottom: $spv-inner--x-small;
     padding-right: $sph-inner;
     padding-top: $spv-inner--x-small;
+  }
+
+  %side-navigation__link {
+    @include vf-focus;
+
+    // vf-highlight-bar is rendered above focus outline
+    // so we need to hide it on focus
+    &:focus::before {
+      display: none;
+    }
+
+    &:hover {
+      text-decoration: none;
+    }
+  }
+
+  .p-side-navigation__link {
+    @extend %side-navigation__link;
+  }
+
+  .p-side-navigation__text,
+  .p-side-navigation__link {
```

Comment:
Can this be merged with the block above?

Reply:
The above is just for `__link` (to have hover and focus), this one shared by for both `__text` and `__link`.